packages/devtools/lib/src/timeline/frame_flame_chart.dart

```diff
@@ -169,13 +198,52 @@ class FrameFlameChart extends CoreElement {
         ..fontWeight = 'bold'
         ..left = '0'
         ..top = '0';
-      section.add(sectionTitle);
+      gpuSection.add(sectionTitle);
 
-      drawRecursively(frame.gpuEventFlow, 0, section);
+      drawRecursively(frame.gpuEventFlow, 0, gpuSection);
     }
 
     drawCpuEvents();
     drawGpuEvents();
+
+    final num scrollWidth = element.scrollWidth;
+
+    // Set the section widths to [scrollWidth], as this should be the max width
+    // we need for the flame chart.
+    cpuSection.element.style.width = '${scrollWidth}px';
+    gpuSection.element.style.width = '${scrollWidth}px';
+
+    if (reRender) {
+      // If we are re-rendering, we need to calculate and set our scroll position.
+      final scrollLeft = scrollWidth *
+          (currentMouseX + currentScrollLeft) /
+          currentScrollWidth -
+          currentMouseX;
+      element.scrollLeft = scrollLeft.round();
+
+      // Maintain the current vertical scrolling position.
+      element.scrollTop = currentScrollTop;
+    } else {
+      // This is the initial render. Calculate our minimum zoom level.
+      if (element.clientWidth == scrollWidth) {
+        // If the entire flame chart fits on the screen, we have no need to zoom
+        // out further.
+        _minZoomLevel = 1;
+      } else {
+        // Multiply by .75 to account for rounding in drawing calculations. This
+        // way the entire frame should fit on the screen at minimum zoom level.
+        _minZoomLevel = min(element.clientWidth / scrollWidth * .75, 1);
+      }
+    }
+  }
+
+  void _reRender() {
```

Comment:
reRender sounds a little strange to me.  Maybe call this method render and split up 
_render into _renderHelper

Reply:
obsolete. removed re-render method.